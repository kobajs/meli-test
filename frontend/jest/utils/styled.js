import { shallow, mount } from "enzyme";
import ThemeProvider from "../../src/theme/ThemeProvider";

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(withTheme(children), options);

  return wrapper;
}

export function withTheme(children) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments);
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments);
}
