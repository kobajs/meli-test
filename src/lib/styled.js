import get from 'lodash/get'

export function theme(path) {
  return props => get(props, `theme.${path}`);
}

export function breakpoints(breakpoint) {
  return props => `@media (min-width: ${props.theme.breakpoints[breakpoint]}px)`;
}