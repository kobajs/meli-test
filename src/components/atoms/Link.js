import React from "react";
import PropTypes from "prop-types";
import NextLink from "next/link";
import styled from "styled-components";

const A = styled.a`
  text-decoration: none;
  color: ${({ color }) => color};
`;

function Link({ children, href, ...props }) {
  return (
    <NextLink href={href}>
      <A href={href} {...props.style}>
        {children}
      </A>
    </NextLink>
  );
}

Link.propTypes = {
  children: PropTypes.node.isRequired,
  href: PropTypes.string,
  style: PropTypes.object
};

Link.defaultProps = {
  style: {
    color: "inherit"
  }
};

export default Link;
