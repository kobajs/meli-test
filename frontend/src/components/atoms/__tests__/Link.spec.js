import React from "react";
import renderer from "react-test-renderer";
// import Router from "next/router";

import Link from "../Link";
import { withTheme } from "../../../../jest/utils/styled";

jest.mock('next/router')

// const mockedRouter = { push: jest.fn(), prefetch: jest.fn() };
// Router.router = mockedRouter;

it("renders correctly", () => {
  const ThemedLink = withTheme(<Link href="/">Test Link</Link>);
  const tree = renderer.create(ThemedLink).toJSON();
  expect(tree).toMatchSnapshot();
});
