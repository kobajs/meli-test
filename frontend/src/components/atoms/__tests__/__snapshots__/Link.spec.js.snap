// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="Link__A-sc-1dh4ev0-0 gsJJXU"
  color="inherit"
  href="/"
  onClick={[Function]}
>
  Test Link
</a>
`;
